void_migration/initial.py

```diff
@@ -38,7 +42,11 @@ def IC(p):
                         remaining, size=int(p.nm * (1 - p.large_concentration) * p.nu_fill), replace=False
                     )
                     s[i, j, small] = p.s_m
-        pre_masked = False
+                    if hasattr(p, "charge_discharge"):
```

Comment:
i dont think this needs to be done inside the loops in lines 34 and 35?

Reply:
done